docs/shared/packages/linter/dependency-checks.md

```diff
@@ -0,0 +1,71 @@
+# Dependency Checks rule
+
+The `@nx/dependency-checks` ESLint rule enables you to discover mismatches between dependencies specified in project's `package.json` and the dependencies that your project depends on.
+
```

Comment:
Need a sentence or two here describing why you would want to use this.  i.e. If your project uses `lodash`, but the `package.json` file does not include `lodash` as a `dependency`, then the project might not work when deployed to an environment where `lodash` isn't already installed.  Enabling this lint rule will catch this problem before your users do.